### PR TITLE
Adding UBI-8 JDK-21 image

### PR DESCRIPTION
**Pull request by matzew**
## Summary

Adding JDK-21 ubi image (see #40)

## Use Cases

New LTS version of java is 21, and we do not yet have that in this repo 

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by pacostas** [comment]
LGTM

**Comment by pacostas** [comment]
@sophiewigmore Can you LGTM ?

**Review by sophiewigmore — APPROVED** [review]

File: metadata_test.go [hunk 1]
```diff
@@ -543,6 +543,69 @@ func testMetadata(t *testing.T, context spec.G, it spec.S) {
 			)))
 		})
 
+		by("confirming that the run java-21 image is correct", func() {
+			dir := filepath.Join(tmpDir, "run-index-java-21")
+			err := os.Mkdir(dir, os.ModePerm)
+			Expect(err).NotTo(HaveOccurred())
+
+			archive, err := os.Open(stack.RunJava17Archive)
```
Comment (pacostas) [inline metadata_test.go:551]:
```suggestion
			archive, err := os.Open(stack.RunJava21Archive)
```

File: metadata_test.go [hunk 2]
```diff
@@ -543,6 +543,69 @@ func testMetadata(t *testing.T, context spec.G, it spec.S) {
...
+				HaveKeyWithValue("io.buildpacks.stack.id", "io.buildpacks.stacks.ubi8"),
+				HaveKeyWithValue("io.buildpacks.stack.description", "ubi8 java-21 image to support buildpacks"),
+				HaveKeyWithValue("io.buildpacks.stack.distro.name", "rhel"),
+				HaveKeyWithValue("io.buildpacks.stack.distro.version", MatchRegexp(`8\.\d+`)),
+				HaveKeyWithValue("io.buildpacks.stack.homepage", "https://github.com/paketo-community/ubi-base-stack"),
+				HaveKeyWithValue("io.buildpacks.stack.maintainer", "Paketo Community"),
+				HaveKeyWithValue("io.buildpacks.stack.metadata", MatchJSON("{}")),
+			))
+
+			runReleaseDateJava17, err := time.Parse(time.RFC3339, file.Config.Labels["io.buildpacks.stack.released"])
```
Comment (pacostas) [inline metadata_test.go:589]:
```suggestion
			runReleaseDateJava21, err := time.Parse(time.RFC3339, file.Config.Labels["io.buildpacks.stack.released"])
```

File: metadata_test.go [hunk 3]
```diff
@@ -543,6 +543,69 @@ func testMetadata(t *testing.T, context spec.G, it spec.S) {
...
+			Expect(err).NotTo(HaveOccurred())
+			Expect(runReleaseDateJava17).NotTo(BeZero())
```
Comment (pacostas) [inline metadata_test.go:591]:
```suggestion
			Expect(runReleaseDateJava21).NotTo(BeZero())
```

**Review by pacostas — CHANGES_REQUESTED** [review]
@matzew Can you also add on .gitignore below line, to exclude the build artifacts from git changes?
`build-java-21/`

By comiting these changes, the tests pass.

**Review by pacostas — APPROVED** [review]
LGTM

**Review by sophiewigmore — APPROVED** [review]

**Comment by matzew** [comment]
@sophiewigmore thanks for approving the PR. Do you think we can merge the PR? 

Once merged / image published, I will update the [ubi-java-extension](https://github.com/paketo-community/ubi-java-extension) repo for leveraging this work here.

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by pacostas** [comment]
@mhdawson this is the next one we need to merge . Also after merging can you close this issue https://github.com/paketo-community/ubi-base-stack/issues/40 ?

**Pull request merged**